### PR TITLE
perf fixes testing

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample2.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                    + " rows");
+            System.out.println(
+                "==============================================================================");
+            break;
+          }
+
+          retryCount++;
+        } while (true);
+        watch.stop();
+        System.out.println("aaaaaaaaa Elapsed Time in Seconds: " + watch.now(TimeUnit.SECONDS));
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample2.java:265]: Nice

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java [hunk 2]
```diff
@@ -737,12 +731,13 @@ private float convertRowToArrow(
 
     // All input values passed validation, iterate over the columns again and combine their existing
     // statistics with the forked statistics for the current row.
-    for (String userInputColumnName : row.keySet()) {
-      String columnName = userInputToUnquotedColumnNameMap.get(userInputColumnName);
-      RowBufferStats stats = statsMap.get(columnName);
-      RowBufferStats forkedStats = forkedStatsMap.get(columnName);
-      statsMap.put(columnName, RowBufferStats.getCombinedStats(stats, forkedStats));
+    for (Map.Entry<String, RowBufferStats> forkedColStats : forkedStatsMap.entrySet()) {
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:734]: I think you can reduce this to `forkedStatsMap.entrySet().forEach(f-> statsMap.put(f.getKey, RowBufferStats.getCombinedStats(statsMap.get(f.getKey), f.getValue()))`

File: src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java [hunk 3]
```diff
@@ -846,14 +842,9 @@ private static String sanitizeValueForExceptionMessage(Object value) {
...
-    CharsetEncoder charsetEncoder =
-        StandardCharsets.UTF_8
-            .newEncoder()
-            .onMalformedInput(CodingErrorAction.REPORT)
-            .onUnmappableCharacter(CodingErrorAction.REPORT);
-    try {
-      charsetEncoder.encode(CharBuffer.wrap(input));
-    } catch (CharacterCodingException e) {
+    byte[] bytes = input.getBytes(StandardCharsets.UTF_8);
+    String roundTripStr = new String(bytes, StandardCharsets.UTF_8);
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java:846]: Nit, inline `bytes`? `new String(input.getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8))`

File: src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample2.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import org.apache.hadoop.util.StopWatch;
+
+/**
+ * Example on how to use the Streaming Ingest client APIs.
+ *
+ * <p>Please read the README.md file for detailed steps
+ */
+public class SnowflakeStreamingIngestExample2 {
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample2.java:29]: Are you planning on checking this in?

Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample2.java:29]: nope, this is for the private jar, as well as for others to take it, the base is 1.1.1, not master

**Pull request closed without merging**